pkgsStatic.libuv fails to build

### Steps To Reproduce
Steps to reproduce the behavior:

```
$ # building from recent master
$ nix-build "https://github.com/NixOS/nixpkgs/archive/23028550cf3064f519bb302361c9500ebc837f92.tar.gz" -A pkgsStatic.libuv

... snip ...

not ok 27 - dlerror
# exit code 134
# Output from process `dlerror`:
# Assertion failed in test/test-dlerror.c on line 46: `strstr(msg, path) != NULL` (0 != 0)
```

See build log for full log

### Full Build log

<details>

```
$ nix-build "https://github.com/NixOS/nixpkgs/archive/23028550cf3064f519bb302361c9500ebc837f92.tar.gz" -A pkgsStatic.libuv
this derivation will be built:
  /nix/store/gcasfqr3rkfhwv3f9yf8iyjf8lpki5mx-libuv-static-x86_64-unknown-linux-musl-1.44.2.drv
building '/nix/store/gcasfqr3rkfhwv3f9yf8iyjf8lpki5mx-libuv-static-x86_64-unknown-linux-musl-1.44.2.drv'...
unpacking sources
unpacking source archive /nix/store/61k994863013giis21bfzwwsf6nr9l87-source
source root is source
patching sources
updateAutotoolsGnuConfigScriptsPhase
configuring
/nix/store/n62963g6w2fj8hnka006wycqwx9pqw5q-gnused-4.8/bin/sed: couldn't flush stdout: Broken pipe
+ libtoolize --copy --force
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
+ aclocal -I m4
+ autoconf
configure.ac:44: warning: The macro `AC_PROG_LIBTOOL' is obsolete.
configure.ac:44: You should run autoupdate.
m4/libtool.m4:100: AC_PROG_LIBTOOL is expanded from...
configure.ac:44: the top level
configure.ac:47: warning: $as_echo is obsolete; use AS_ECHO(["message"]) instead
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
m4/ax_pthread.m4:88: AX_PTHREAD is expanded from...
configure.ac:47: the top level
+ automake --add-missing --copy
configure.ac:42: installing './ar-lib'
configure.ac:25: installing './compile'
configure.ac:22: installing './config.guess'
configure.ac:22: installing './config.sub'
configure.ac:21: installing './install-sh'
configure.ac:21: installing './missing'
Makefile.am: installing './depcomp'
fixing libtool script ./ltmain.sh
./configure
configure flags: --disable-dependency-tracking --prefix=/nix/store/43dqvzf3v2hy9h0irlyajc4wbwqxns4w-libuv-static-x86_64-unknown-linux-musl-1.44.2 --enable-static --disable-shared --disable-shared --build=x86_64-unknown-linux-gnu --host=x86_64-unknown-linux-musl
checking for a BSD-compatible install... /nix/store/mj6wiv9gbbydh06x2i4fl5a9ksdck7c5-coreutils-9.1/bin/install -c
checking whether build environment is sane... yes
checking for x86_64-unknown-linux-musl-strip... x86_64-unknown-linux-musl-strip
checking for a race-free mkdir -p... /nix/store/mj6wiv9gbbydh06x2i4fl5a9ksdck7c5-coreutils-9.1/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-musl
checking for x86_64-unknown-linux-musl-gcc... x86_64-unknown-linux-musl-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether x86_64-unknown-linux-musl-gcc accepts -g... yes
checking for x86_64-unknown-linux-musl-gcc option to enable C11 features... none needed
checking whether x86_64-unknown-linux-musl-gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of x86_64-unknown-linux-musl-gcc... none
checking for x86_64-unknown-linux-musl-gcc way to treat warnings as errors... -Werror
checking if x86_64-unknown-linux-musl-gcc supports __attribute__(( visibility("default") ))... yes
checking if x86_64-unknown-linux-musl-gcc supports -fvisibility=hidden... yes
checking if x86_64-unknown-linux-musl-gcc supports -fno-strict-aliasing flag... yes
checking if x86_64-unknown-linux-musl-gcc supports -fno-strict-aliasing flag... yes
checking if x86_64-unknown-linux-musl-gcc supports -g flag... yes
checking if x86_64-unknown-linux-musl-gcc supports -std=gnu89 flag... yes
checking if x86_64-unknown-linux-musl-gcc supports -Wall flag... yes
checking if x86_64-unknown-linux-musl-gcc supports -Wextra flag... yes
checking if x86_64-unknown-linux-musl-gcc supports -Wno-long-long flag... yes
checking if x86_64-unknown-linux-musl-gcc supports -Wno-unused-parameter flag... yes
checking if x86_64-unknown-linux-musl-gcc supports -Wstrict-prototypes flag... yes
checking for x86_64-unknown-linux-musl-ar... x86_64-unknown-linux-musl-ar
checking the archiver (x86_64-unknown-linux-musl-ar) interface... ar
checking how to print strings... printf
checking for a sed that does not truncate output... /nix/store/n62963g6w2fj8hnka006wycqwx9pqw5q-gnused-4.8/bin/sed
checking for grep that handles long lines and -e... /nix/store/zpjn33mx0pbfc71ddfi9sc8pdfqrsh8z-gnugrep-3.7/bin/grep
checking for egrep... /nix/store/zpjn33mx0pbfc71ddfi9sc8pdfqrsh8z-gnugrep-3.7/bin/grep -E
checking for fgrep... /nix/store/zpjn33mx0pbfc71ddfi9sc8pdfqrsh8z-gnugrep-3.7/bin/grep -F
checking for ld used by x86_64-unknown-linux-musl-gcc... x86_64-unknown-linux-musl-ld
checking if the linker (x86_64-unknown-linux-musl-ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... x86_64-unknown-linux-musl-nm
checking the name lister (x86_64-unknown-linux-musl-nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 32768
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-musl format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for x86_64-unknown-linux-musl-ld option to reload object files... -r
checking for x86_64-unknown-linux-musl-file... no
checking for file... file
configure: WARNING: using cross tools not prefixed with host triplet
checking for x86_64-unknown-linux-musl-objdump... x86_64-unknown-linux-musl-objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for x86_64-unknown-linux-musl-dlltool... no
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for x86_64-unknown-linux-musl-ar... (cached) x86_64-unknown-linux-musl-ar
checking for archiver @FILE support... @
checking for x86_64-unknown-linux-musl-strip... (cached) x86_64-unknown-linux-musl-strip
checking for x86_64-unknown-linux-musl-ranlib... x86_64-unknown-linux-musl-ranlib
checking command to parse x86_64-unknown-linux-musl-nm output from x86_64-unknown-linux-musl-gcc object... ok
checking for sysroot... no
checking for a working dd... /nix/store/mj6wiv9gbbydh06x2i4fl5a9ksdck7c5-coreutils-9.1/bin/dd
checking how to truncate binary pipes... /nix/store/mj6wiv9gbbydh06x2i4fl5a9ksdck7c5-coreutils-9.1/bin/dd bs=4096 count=1
checking for x86_64-unknown-linux-musl-mt... no
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if x86_64-unknown-linux-musl-gcc supports -fno-rtti -fno-exceptions... no
checking for x86_64-unknown-linux-musl-gcc option to produce PIC... -fPIC -DPIC
checking if x86_64-unknown-linux-musl-gcc PIC flag -fPIC -DPIC works... yes
checking if x86_64-unknown-linux-musl-gcc static flag -static works... yes
checking if x86_64-unknown-linux-musl-gcc supports -c -o file.o... yes
checking if x86_64-unknown-linux-musl-gcc supports -c -o file.o... (cached) yes
checking whether the x86_64-unknown-linux-musl-gcc linker (x86_64-unknown-linux-musl-ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking whether make supports nested variables... (cached) yes
checking how to run the C preprocessor... x86_64-unknown-linux-musl-gcc -E
checking whether x86_64-unknown-linux-musl-gcc is Clang... no
checking whether pthreads work with -pthread... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking for library containing dlopen... none required
checking for library containing kstat_lookup... no
checking for library containing gethostbyname... none required
checking for library containing perfstat_cpu... no
checking for library containing clock_gettime... none required
checking for library containing sendfile... none required
checking for library containing socket... none required
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for sys/ahafs_evProds.h... no
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating libuv.pc
config.status: executing depfiles commands
config.status: executing libtool commands
building
build flags: -j12 -l12 SHELL=/nix/store/1b9p07z77phvv2hf6gm9f28syp39f1ag-bash-5.1-p16/bin/bash  CC       src/libuv_la-fs-poll.lo
  CC       src/libuv_la-idna.lo
  CC       src/libuv_la-inet.lo
  CC       src/libuv_la-random.lo
  CC       src/libuv_la-strscpy.lo
  CC       src/libuv_la-threadpool.lo
  CC       src/libuv_la-timer.lo
  CC       src/libuv_la-uv-data-getter-setters.lo
  CC       src/libuv_la-uv-common.lo
  CC       src/libuv_la-version.lo
  CC       src/libuv_la-strtok.lo
  CC       src/unix/libuv_la-async.lo
  CC       src/unix/libuv_la-core.lo
  CC       src/unix/libuv_la-dl.lo
  CC       src/unix/libuv_la-fs.lo
  CC       src/unix/libuv_la-getaddrinfo.lo
  CC       src/unix/libuv_la-getnameinfo.lo
  CC       src/unix/libuv_la-loop-watcher.lo
  CC       src/unix/libuv_la-loop.lo
  CC       src/unix/libuv_la-pipe.lo
  CC       src/unix/libuv_la-poll.lo
  CC       src/unix/libuv_la-process.lo
  CC       src/unix/libuv_la-random-devurandom.lo
  CC       src/unix/libuv_la-signal.lo
  CC       src/unix/libuv_la-stream.lo
  CC       src/unix/libuv_la-tcp.lo
  CC       src/unix/libuv_la-thread.lo
  CC       src/unix/libuv_la-tty.lo
  CC       src/unix/libuv_la-udp.lo
  CC       src/unix/libuv_la-linux-core.lo
  CC       src/unix/libuv_la-linux-inotify.lo
  CC       src/unix/libuv_la-linux-syscalls.lo
  CC       src/unix/libuv_la-procfs-exepath.lo
  CC       src/unix/libuv_la-proctitle.lo
  CC       src/unix/libuv_la-random-getrandom.lo
  CC       src/unix/libuv_la-random-sysctl-linux.lo
  CC       src/unix/libuv_la-epoll.lo
  CCLD     libuv.la
running tests
check flags: -j12 -l12 SHELL=/nix/store/1b9p07z77phvv2hf6gm9f28syp39f1ag-bash-5.1-p16/bin/bash VERBOSE=y check
make  test/run-tests
make[1]: Entering directory '/build/source'
  CC       test/run_tests-blackhole-server.o
  CC       test/run_tests-echo-server.o
  CC       test/run_tests-run-tests.o
  CC       test/run_tests-runner.o
  CC       test/run_tests-test-active.o
  CC       test/run_tests-test-async.o
  CC       test/run_tests-test-async-null-cb.o
  CC       test/run_tests-test-barrier.o
  CC       test/run_tests-test-callback-stack.o
  CC       test/run_tests-test-close-fd.o
  CC       test/run_tests-test-close-order.o
  CC       test/run_tests-test-condvar.o
  CC       test/run_tests-test-connect-unspecified.o
  CC       test/run_tests-test-connection-fail.o
  CC       test/run_tests-test-cwd-and-chdir.o
  CC       test/run_tests-test-default-loop-close.o
  CC       test/run_tests-test-delayed-accept.o
  CC       test/run_tests-test-dlerror.o
  CC       test/run_tests-test-eintr-handling.o
  CC       test/run_tests-test-embed.o
  CC       test/run_tests-test-emfile.o
  CC       test/run_tests-test-env-vars.o
  CC       test/run_tests-test-error.o
  CC       test/run_tests-test-fail-always.o
  CC       test/run_tests-test-fs-copyfile.o
  CC       test/run_tests-test-fs-event.o
  CC       test/run_tests-test-fs-poll.o
  CC       test/run_tests-test-fs.o
  CC       test/run_tests-test-fs-readdir.o
  CC       test/run_tests-test-fs-fd-hash.o
  CC       test/run_tests-test-fs-open-flags.o
  CC       test/run_tests-test-fork.o
  CC       test/run_tests-test-getters-setters.o
  CC       test/run_tests-test-get-currentexe.o
  CC       test/run_tests-test-get-loadavg.o
  CC       test/run_tests-test-get-memory.o
  CC       test/run_tests-test-get-passwd.o
  CC       test/run_tests-test-getaddrinfo.o
  CC       test/run_tests-test-gethostname.o
  CC       test/run_tests-test-getnameinfo.o
  CC       test/run_tests-test-getsockname.o
  CC       test/run_tests-test-gettimeofday.o
  CC       test/run_tests-test-handle-fileno.o
  CC       test/run_tests-test-homedir.o
  CC       test/run_tests-test-hrtime.o
  CC       test/run_tests-test-idle.o
  CC       test/run_tests-test-idna.o
  CC       test/run_tests-test-ip4-addr.o
  CC       test/run_tests-test-ip6-addr.o
  CC       test/run_tests-test-ip-name.o
  CC       test/run_tests-test-ipc-heavy-traffic-deadlock-bug.o
  CC       test/run_tests-test-ipc-send-recv.o
  CC       test/run_tests-test-ipc.o
  CC       test/run_tests-test-loop-handles.o
  CC       test/run_tests-test-loop-alive.o
  CC       test/run_tests-test-loop-close.o
  CC       test/run_tests-test-loop-stop.o
  CC       test/run_tests-test-loop-time.o
  CC       test/run_tests-test-loop-configure.o
  CC       test/run_tests-test-metrics.o
  CC       test/run_tests-test-multiple-listen.o
  CC       test/run_tests-test-mutexes.o
  CC       test/run_tests-test-not-readable-nor-writable-on-read-error.o
  CC       test/run_tests-test-not-writable-after-shutdown.o
  CC       test/run_tests-test-osx-select.o
  CC       test/run_tests-test-pass-always.o
  CC       test/run_tests-test-ping-pong.o
  CC       test/run_tests-test-pipe-bind-error.o
  CC       test/run_tests-test-pipe-connect-error.o
  CC       test/run_tests-test-pipe-connect-multiple.o
  CC       test/run_tests-test-pipe-connect-prepare.o
  CC       test/run_tests-test-pipe-getsockname.o
  CC       test/run_tests-test-pipe-pending-instances.o
  CC       test/run_tests-test-pipe-sendmsg.o
  CC       test/run_tests-test-pipe-server-close.o
  CC       test/run_tests-test-pipe-close-stdout-read-stdin.o
  CC       test/run_tests-test-pipe-set-non-blocking.o
  CC       test/run_tests-test-pipe-set-fchmod.o
  CC       test/run_tests-test-platform-output.o
  CC       test/run_tests-test-poll.o
  CC       test/run_tests-test-poll-close.o
  CC       test/run_tests-test-poll-close-doesnt-corrupt-stack.o
  CC       test/run_tests-test-poll-closesocket.o
  CC       test/run_tests-test-poll-multiple-handles.o
  CC       test/run_tests-test-poll-oob.o
  CC       test/run_tests-test-process-priority.o
  CC       test/run_tests-test-process-title.o
  CC       test/run_tests-test-process-title-threadsafe.o
  CC       test/run_tests-test-queue-foreach-delete.o
  CC       test/run_tests-test-random.o
  CC       test/run_tests-test-readable-on-eof.o
  CC       test/run_tests-test-ref.o
  CC       test/run_tests-test-run-nowait.o
  CC       test/run_tests-test-run-once.o
  CC       test/run_tests-test-semaphore.o
  CC       test/run_tests-test-shutdown-close.o
  CC       test/run_tests-test-shutdown-eof.o
  CC       test/run_tests-test-shutdown-simultaneous.o
  CC       test/run_tests-test-shutdown-twice.o
  CC       test/run_tests-test-signal-multiple-loops.o
  CC       test/run_tests-test-signal-pending-on-close.o
  CC       test/run_tests-test-signal.o
  CC       test/run_tests-test-socket-buffer-size.o
  CC       test/run_tests-test-spawn.o
  CC       test/run_tests-test-stdio-over-pipes.o
  CC       test/run_tests-test-strscpy.o
  CC       test/run_tests-test-strtok.o
  CC       test/run_tests-test-tcp-alloc-cb-fail.o
  CC       test/run_tests-test-tcp-bind-error.o
  CC       test/run_tests-test-tcp-bind6-error.o
  CC       test/run_tests-test-tcp-close-accept.o
  CC       test/run_tests-test-tcp-close-while-connecting.o
  CC       test/run_tests-test-tcp-close-after-read-timeout.o
  CC       test/run_tests-test-tcp-close.o
  CC       test/run_tests-test-tcp-close-reset.o
  CC       test/run_tests-test-tcp-create-socket-early.o
  CC       test/run_tests-test-tcp-connect-error-after-write.o
  CC       test/run_tests-test-tcp-connect-error.o
  CC       test/run_tests-test-tcp-connect-timeout.o
  CC       test/run_tests-test-tcp-connect6-error.o
  CC       test/run_tests-test-tcp-flags.o
  CC       test/run_tests-test-tcp-open.o
  CC       test/run_tests-test-tcp-read-stop.o
  CC       test/run_tests-test-tcp-read-stop-start.o
  CC       test/run_tests-test-tcp-rst.o
  CC       test/run_tests-test-tcp-shutdown-after-write.o
  CC       test/run_tests-test-tcp-unexpected-read.o
  CC       test/run_tests-test-tcp-oob.o
  CC       test/run_tests-test-tcp-write-to-half-open-connection.o
  CC       test/run_tests-test-tcp-write-after-connect.o
  CC       test/run_tests-test-tcp-writealot.o
  CC       test/run_tests-test-tcp-write-fail.o
  CC       test/run_tests-test-tcp-try-write.o
  CC       test/run_tests-test-tcp-try-write-error.o
  CC       test/run_tests-test-tcp-write-queue-order.o
  CC       test/run_tests-test-test-macros.o
  CC       test/run_tests-test-thread-equal.o
  CC       test/run_tests-test-thread.o
  CC       test/run_tests-test-threadpool-cancel.o
  CC       test/run_tests-test-threadpool.o
  CC       test/run_tests-test-timer-again.o
  CC       test/run_tests-test-timer-from-check.o
  CC       test/run_tests-test-timer.o
  CC       test/run_tests-test-tmpdir.o
  CC       test/run_tests-test-tty-duplicate-key.o
  CC       test/run_tests-test-tty-escape-sequence-processing.o
  CC       test/run_tests-test-tty.o
  CC       test/run_tests-test-udp-alloc-cb-fail.o
  CC       test/run_tests-test-udp-bind.o
  CC       test/run_tests-test-udp-connect.o
  CC       test/run_tests-test-udp-connect6.o
  CC       test/run_tests-test-udp-create-socket-early.o
  CC       test/run_tests-test-udp-dgram-too-big.o
  CC       test/run_tests-test-udp-ipv6.o
  CC       test/run_tests-test-udp-mmsg.o
  CC       test/run_tests-test-udp-multicast-interface.o
  CC       test/run_tests-test-udp-multicast-interface6.o
  CC       test/run_tests-test-udp-multicast-join.o
  CC       test/run_tests-test-udp-multicast-join6.o
  CC       test/run_tests-test-udp-multicast-ttl.o
  CC       test/run_tests-test-udp-open.o
  CC       test/run_tests-test-udp-options.o
  CC       test/run_tests-test-udp-send-and-recv.o
  CC       test/run_tests-test-udp-send-hang-loop.o
  CC       test/run_tests-test-udp-send-immediate.o
  CC       test/run_tests-test-udp-sendmmsg-error.o
  CC       test/run_tests-test-udp-send-unreachable.o
  CC       test/run_tests-test-udp-try-send.o
  CC       test/run_tests-test-uname.o
  CC       test/run_tests-test-walk-handles.o
  CC       test/run_tests-test-watcher-cross-stop.o
  CC       test/run_tests-runner-unix.o
  CCLD     test/run-tests
make[1]: Leaving directory '/build/source'
make  check-TESTS
make[1]: Entering directory '/build/source'
1..394
ok 1 - platform_output
# Output from process `platform_output`:
# uv_get_process_title: ./test/run-tests
# uv_cwd: /build/source
# uv_resident_set_memory: 409600
# uv_uptime: 3722.710000
# uv_getrusage:
#   user: 0 sec 288 microsec
#   system: 0 sec 0 microsec
#   page faults: 0
#   maximum resident set size: 400
# uv_available_parallelism: 12
# uv_cpu_info:
#   model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
#   speed: 2200
#   times.sys: 11700
#   times.user: 20070
#   times.idle: 1932290
#   times.irq: 0
#   times.nice: 260
#   model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
#   speed: 2200
#   times.sys: 9800
#   times.user: 24180
#   times.idle: 917640
#   times.irq: 0
#   times.nice: 440
#   model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
#   speed: 2200
#   times.sys: 8060
#   times.user: 23470
#   times.idle: 917700
#   times.irq: 0
#   times.nice: 360
#   model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
#   speed: 2200
#   times.sys: 12400
#   times.user: 30770
#   times.idle: 911140
#   times.irq: 0
#   times.nice: 240
#   model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
#   speed: 2200
#   times.sys: 17510
#   times.user: 25700
#   times.idle: 91777
# 0
#   times.irq: 0
#   times.nice: 460
#   model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
#   speed: 2200
#   times.sys: 10000
#   times.user: 26800
#   times.idle: 917670
#   times.irq: 0
#   times.nice: 380
#   model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
#   speed: 2200
#   times.sys: 6760
#   times.user: 17150
#   times.idle: 922270
#   times.irq: 0
#   times.nice: 560
#   model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
#   speed: 2200
#   times.sys: 8190
#   times.user: 22830
#   times.idle: 921470
#   times.irq: 0
#   times.nice: 590
#   model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
#   speed: 2200
#   times.sys: 7650
#   times.user: 25640
#   times.idle: 916990
#   times.irq: 0
#   times.nice: 250
#   model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
#   speed: 2200
#   times.sys: 8940
#   times.user: 21340
#   times.idle: 918590
#   times.irq: 0
#   times.nice: 860
#   model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
#   speed: 3134
#   times.sys: 9840
#   times.user: 23470
#   times.idle: 922080
#   times.irq: 0
#   times.nice: 1210
#   model: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
#  
#  speed: 2200
#   times.sys: 8840
#   times.user: 26770
#   times.idle: 919000
#   times.irq: 0
#   times.nice: 1670
# uv_interface_addresses:
#   name: lo
#   internal: 1
#   physical address: 00:00:00:00:00:00
#   address: 127.0.0.1
#   netmask: 255.0.0.0
#   name: lo
#   internal: 1
#   physical address: 00:00:00:00:00:00
#   address: ::1
#   netmask: ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff
# uv_os_get_passwd:
#   euid: 1000
#   gid: 100
#   username: nixbld
#   shell: /noshell
#   home directory: /build
# uv_os_getpid: 7144
# uv_os_getppid: 7143
# uv_os_uname:
#   sysname: Linux
#   release: 5.15.63
#   version: #1-NixOS SMP Thu Aug 25 09:40:49 UTC 2022
#   machine: x86_64
ok 2 - active
ok 3 - async
ok 4 - async_null_cb
ok 5 - async_ref
ok 6 - barrier_1
ok 7 - barrier_2
ok 8 - barrier_3
ok 9 - barrier_serial_thread
ok 10 - barrier_serial_thread_single
ok 11 - callback_stack
ok 12 - check_ref
ok 13 - close_fd
ok 14 - close_order
ok 15 - closed_fd_events
ok 16 - condvar_1
ok 17 - condvar_2
ok 18 - condvar_3
ok 19 - condvar_4
ok 20 - condvar_5
ok 21 - connect_unspecified
ok 22 - connection_fail
ok 23 - connection_fail_doesnt_auto_close
ok 24 - cwd_and_chdir
ok 25 - default_loop_close
ok 26 - delayed_accept
not ok 27 - dlerror
# exit code 134
# Output from process `dlerror`:
# Assertion failed in test/test-dlerror.c on line 46: `strstr(msg, path) != NULL` (0 != 0)
ok 28 - eintr_handling
ok 29 - embed
ok 30 - emfile
ok 31 - env_vars
ok 32 - error_message
ok 33 - fork_fs_events_child
ok 34 - fork_fs_events_child_dir
ok 35 - fork_fs_events_file_parent_child
ok 36 - fork_signal_to_child
ok 37 - fork_signal_to_child_closed
ok 38 - fork_socketpair
ok 39 - fork_socketpair_started
ok 40 - fork_threadpool_queue_work_simple
ok 41 - fork_timer
ok 42 - fs_access
ok 43 - fs_async_dir
ok 44 - fs_async_sendfile
ok 45 - fs_async_sendfile_nodata
ok 46 - fs_chmod
ok 47 - fs_copyfile
ok 48 - fs_event_close_in_callback
ok 49 - fs_event_close_with_pending_delete_event
ok 50 - fs_event_close_with_pending_event
ok 51 - fs_event_error_reporting
ok 52 - fs_event_getpath
ok 53 - fs_event_immediate_close
ok 54 - fs_event_no_callback_after_close
ok 55 - fs_event_no_callback_on_close
ok 56 - fs_event_ref
ok 57 - fs_event_start_and_close
ok 58 - fs_event_stop_in_cb
ok 59 - fs_event_watch_dir
ok 60 - fs_event_watch_dir_recursive # SKIP Recursive directory watching not supported on this platform.
ok 61 - fs_event_watch_file
ok 62 - fs_event_watch_file_current_dir
ok 63 - fs_event_watch_file_exact_path
ok 64 - fs_event_watch_file_twice
ok 65 - fs_event_watch_invalid_path
ok 66 - fs_file_async
ok 67 - fs_file_loop
ok 68 - fs_file_nametoolong
ok 69 - fs_file_noent
ok 70 - fs_file_open_append
ok 71 - fs_file_pos_after_op_with_offset
ok 72 - fs_file_sync
ok 73 - fs_file_write_null_buffer
ok 74 - fs_futime
ok 75 - fs_get_system_error
ok 76 - fs_lutime
ok 77 - fs_mkdtemp
ok 78 - fs_mkstemp
ok 79 - fs_null_req
ok 80 - fs_open_dir
ok 81 - fs_partial_read
ok 82 - fs_partial_write
ok 83 - fs_poll
ok 84 - fs_poll_close_request
ok 85 - fs_poll_close_request_multi_start_stop
ok 86 - fs_poll_close_request_multi_stop_start
ok 87 - fs_poll_close_request_stop_when_active
ok 88 - fs_poll_getpath
ok 89 - fs_poll_ref
ok 90 - fs_read_bufs
ok 91 - fs_read_dir
ok 92 - fs_read_file_eof
ok 93 - fs_read_write_null_arguments
ok 94 - fs_readdir_empty_dir
ok 95 - fs_readdir_file
ok 96 - fs_readdir_non_empty_dir
ok 97 - fs_readdir_non_existing_dir
ok 98 - fs_readlink
ok 99 - fs_realpath
ok 100 - fs_rename_to_existing_file
ok 101 - fs_scandir_empty_dir
ok 102 - fs_scandir_file
ok 103 - fs_scandir_non_existent_dir
ok 104 - fs_stat_missing_path
ok 105 - fs_statfs
ok 106 - fs_symlink
ok 107 - fs_symlink_dir
ok 108 - fs_unlink_readonly
ok 109 - fs_utime
ok 110 - fs_utime_round
ok 111 - fs_write_alotof_bufs
ok 112 - fs_write_alotof_bufs_with_offset
ok 113 - fs_write_multiple_bufs
ok 114 - get_currentexe
ok 115 - get_loadavg
ok 116 - get_memory
ok 117 - get_osfhandle_valid_handle
ok 118 - getaddrinfo_basic
ok 119 - getaddrinfo_basic_sync
ok 120 - getaddrinfo_concurrent
ok 121 - gethostname
ok 122 - getsockname_tcp
ok 123 - getsockname_udp
ok 124 - getters_setters
ok 125 - gettimeofday
ok 126 - handle_fileno
ok 127 - handle_type_name
ok 128 - has_ref
ok 129 - homedir
ok 130 - hrtime
ok 131 - idle_check
ok 132 - idle_ref
ok 133 - idle_starvation
ok 134 - idna_toascii
ok 135 - ip4_addr
ok 136 - ip6_addr_link_local
ok 137 - ip6_pton
ok 138 - ip6_sin6_len
ok 139 - ip_name
ok 140 - ipc_heavy_traffic_deadlock_bug
ok 141 - ipc_listen_after_write
ok 142 - ipc_listen_before_write
ok 143 - ipc_send_recv_pipe
ok 144 - ipc_send_recv_pipe_inprocess
ok 145 - ipc_send_recv_tcp
ok 146 - ipc_send_recv_tcp_inprocess
ok 147 - ipc_send_zero
ok 148 - ipc_tcp_connection
ok 149 - kill
ok 150 - kill_invalid_signum
ok 151 - loop_alive
ok 152 - loop_backend_timeout
ok 153 - loop_close
ok 154 - loop_configure
ok 155 - loop_handles
ok 156 - loop_instant_close
ok 157 - loop_stop
ok 158 - loop_update_time
ok 159 - metrics_idle_time
ok 160 - metrics_idle_time_thread
ok 161 - metrics_idle_time_zero
ok 162 - multiple_listen
ok 163 - not_readable_nor_writable_on_read_error
ok 164 - not_writable_after_shutdown
ok 165 - open_osfhandle_valid_handle
ok 166 - pipe_bind_error_addrinuse
ok 167 - pipe_bind_error_addrnotavail
ok 168 - pipe_bind_error_inval
ok 169 - pipe_bind_or_listen_error_after_close
ok 170 - pipe_close_stdout_read_stdin
ok 171 - pipe_connect_bad_name
ok 172 - pipe_connect_multiple
ok 173 - pipe_connect_on_prepare
ok 174 - pipe_connect_to_file
ok 175 - pipe_getsockname
ok 176 - pipe_getsockname_abstract
ok 177 - pipe_getsockname_blocking
ok 178 - pipe_listen_without_bind
ok 179 - pipe_pending_instances
ok 180 - pipe_ping_pong
ok 181 - pipe_ping_pong_vec
ok 182 - pipe_ref
ok 183 - pipe_ref2
ok 184 - pipe_ref3
ok 185 - pipe_ref4
ok 186 - pipe_sendmsg
ok 187 - pipe_server_close
ok 188 - pipe_set_chmod
ok 189 - pipe_set_non_blocking
ok 190 - poll_bad_fdtype
ok 191 - poll_close
ok 192 - poll_close_doesnt_corrupt_stack # SKIP Test only relevant on Windows
ok 193 - poll_closesocket # SKIP Test only relevant on Windows
ok 194 - poll_duplex
ok 195 - poll_multiple_handles
ok 196 - poll_nested_epoll
ok 197 - poll_oob
ok 198 - poll_unidirectional
ok 199 - prepare_ref
ok 200 - process_priority
ok 201 - process_ref
ok 202 - process_title
ok 203 - process_title_big_argv
ok 204 - process_title_threadsafe
ok 205 - queue_foreach_delete
ok 206 - random_async
ok 207 - random_sync
ok 208 - readable_on_eof
ok 209 - ref
ok 210 - req_type_name
ok 211 - run_nowait
ok 212 - run_once
ok 213 - semaphore_1
ok 214 - semaphore_2
ok 215 - semaphore_3
ok 216 - shutdown_close_pipe
ok 217 - shutdown_close_tcp
ok 218 - shutdown_eof
ok 219 - shutdown_simultaneous
ok 220 - shutdown_twice
ok 221 - signal_close_loop_alive
ok 222 - signal_multiple_loops
ok 223 - signal_pending_on_close
ok 224 - socket_buffer_size
ok 225 - spawn_and_kill
ok 226 - spawn_and_kill_with_std
ok 227 - spawn_and_ping
ok 228 - spawn_auto_unref
ok 229 - spawn_closed_process_io
ok 230 - spawn_detached
ok 231 - spawn_empty_env
ok 232 - spawn_exercise_sigchld_issue
ok 233 - spawn_exit_code
ok 234 - spawn_fails
ok 235 - spawn_fails_check_for_waitpid_cleanup
ok 236 - spawn_fs_open
ok 237 - spawn_ignored_stdio
ok 238 - spawn_inherit_streams
ok 239 - spawn_preserve_env
ok 240 - spawn_quoted_path # SKIP Test for Windows
ok 241 - spawn_reads_child_path
ok 242 - spawn_relative_path
ok 243 - spawn_setuid_setgid # SKIP It should be run as root user
ok 244 - spawn_stdin
ok 245 - spawn_stdio_greater_than_3
ok 246 - spawn_stdout
ok 247 - spawn_stdout_and_stderr_to_file
ok 248 - spawn_stdout_and_stderr_to_file2
ok 249 - spawn_stdout_and_stderr_to_file_swap
ok 250 - spawn_stdout_to_file
ok 251 - spawn_tcp_server
ok 252 - stdio_emulate_iocp
ok 253 - stdio_over_pipes
ok 254 - strscpy
ok 255 - strtok
ok 256 - sys_error
ok 257 - tcp6_local_connect_timeout
ok 258 - tcp6_ping_pong
ok 259 - tcp6_ping_pong_vec
ok 260 - tcp_alloc_cb_fail
ok 261 - tcp_bind6_error_addrinuse
ok 262 - tcp_bind6_error_addrnotavail
ok 263 - tcp_bind6_error_fault
ok 264 - tcp_bind6_error_inval
ok 265 - tcp_bind6_localhost_ok
ok 266 - tcp_bind_error_addrinuse_connect
ok 267 - tcp_bind_error_addrinuse_listen
ok 268 - tcp_bind_error_addrnotavail_1
ok 269 - tcp_bind_error_addrnotavail_2
ok 270 - tcp_bind_error_fault
ok 271 - tcp_bind_error_inval
ok 272 - tcp_bind_invalid_flags
ok 273 - tcp_bind_localhost_ok
ok 274 - tcp_bind_or_listen_error_after_close
ok 275 - tcp_bind_writable_flags
ok 276 - tcp_close
ok 277 - tcp_close_accept
ok 278 - tcp_close_after_read_timeout
ok 279 - tcp_close_reset_accepted
ok 280 - tcp_close_reset_accepted_after_shutdown
ok 281 - tcp_close_reset_accepted_after_socket_shutdown
ok 282 - tcp_close_reset_client
ok 283 - tcp_close_reset_client_after_shutdown
ok 284 - tcp_close_while_connecting # SKIP Network unreachable.
ok 285 - tcp_connect_error_after_write
ok 286 - tcp_connect_error_fault
ok 287 - tcp_create_early
ok 288 - tcp_create_early_accept
ok 289 - tcp_create_early_bad_bind
ok 290 - tcp_create_early_bad_domain
ok 291 - tcp_flags
ok 292 - tcp_listen_without_bind
ok 293 - tcp_local_connect_timeout
ok 294 - tcp_oob
ok 295 - tcp_open
ok 296 - tcp_open_bound
ok 297 - tcp_open_connected
ok 298 - tcp_open_twice
ok 299 - tcp_ping_pong
ok 300 - tcp_ping_pong_vec
ok 301 - tcp_read_stop
ok 302 - tcp_read_stop_start
ok 303 - tcp_ref
ok 304 - tcp_ref2
ok 305 - tcp_ref2b
ok 306 - tcp_ref3
ok 307 - tcp_ref4
ok 308 - tcp_rst
ok 309 - tcp_shutdown_after_write
ok 310 - tcp_try_write
ok 311 - tcp_try_write_error
ok 312 - tcp_unexpected_read
ok 313 - tcp_write_after_connect
ok 314 - tcp_write_fail
ok 315 - tcp_write_queue_order
ok 316 - tcp_write_ready
ok 317 - tcp_write_to_half_open_connection
ok 318 - test_macros
ok 319 - thread_create
ok 320 - thread_equal
ok 321 - thread_local_storage
ok 322 - thread_mutex
ok 323 - thread_mutex_recursive
ok 324 - thread_rwlock
ok 325 - thread_rwlock_trylock
ok 326 - thread_stack_size
ok 327 - thread_stack_size_explicit
ok 328 - threadpool_cancel_fs
ok 329 - threadpool_cancel_getaddrinfo
ok 330 - threadpool_cancel_getnameinfo
ok 331 - threadpool_cancel_random
ok 332 - threadpool_cancel_single
ok 333 - threadpool_cancel_work
ok 334 - threadpool_queue_work_einval
ok 335 - threadpool_queue_work_simple
ok 336 - timer
ok 337 - timer_again
ok 338 - timer_early_check
ok 339 - timer_from_check
ok 340 - timer_huge_repeat
ok 341 - timer_huge_timeout
ok 342 - timer_init
ok 343 - timer_is_closing
ok 344 - timer_null_callback
ok 345 - timer_order
ok 346 - timer_ref
ok 347 - timer_ref2
ok 348 - timer_run_once
ok 349 - timer_start_twice
ok 350 - tmpdir
ok 351 - tty # SKIP Cannot open /dev/tty as read-only: No such device or address
ok 352 - tty_file
ok 353 - tty_pty
ok 354 - udp_alloc_cb_fail
ok 355 - udp_bind
ok 356 - udp_bind_reuseaddr
ok 357 - udp_connect
ok 358 - udp_connect6
ok 359 - udp_create_early
ok 360 - udp_create_early_bad_bind
ok 361 - udp_create_early_bad_domain
ok 362 - udp_dgram_too_big
ok 363 - udp_dual_stack
ok 364 - udp_ipv6_only
ok 365 - udp_mmsg
ok 366 - udp_multicast_interface
ok 367 - udp_multicast_interface6
ok 368 - udp_multicast_ttl
ok 369 - udp_no_autobind
ok 370 - udp_open
ok 371 - udp_open_bound
ok 372 - udp_open_connect
ok 373 - udp_open_twice
ok 374 - udp_options
ok 375 - udp_options6
ok 376 - udp_ref
ok 377 - udp_ref2
ok 378 - udp_ref3
ok 379 - udp_send_and_recv
ok 380 - udp_send_immediate
ok 381 - udp_send_unix
ok 382 - udp_send_unreachable
ok 383 - udp_sendmmsg_error
ok 384 - udp_try_send
ok 385 - uname
ok 386 - unref_in_prepare_cb
ok 387 - utf8_decode1
ok 388 - utf8_decode1_overrun
ok 389 - walk_handles
ok 390 - watcher_cross_stop
ok 391 - we_get_signal
ok 392 - we_get_signal_one_shot
ok 393 - we_get_signals
ok 394 - we_get_signals_mixed
FAIL: test/run-tests
======================================================
1 of 1 test failed
Please report to https://github.com/libuv/libuv/issues
======================================================
make[1]: *** [Makefile:5613: check-TESTS] Error 1
make[1]: Leaving directory '/build/source'
make: *** [Makefile:5878: check-am] Error 2
error: builder for '/nix/store/gcasfqr3rkfhwv3f9yf8iyjf8lpki5mx-libuv-static-x86_64-unknown-linux-musl-1.44.2.drv' failed with exit code 2;
       last 10 log lines:
       > ok 393 - we_get_signals
       > ok 394 - we_get_signals_mixed
       > FAIL: test/run-tests
       > ======================================================
       > 1 of 1 test failed
       > Please report to https://github.com/libuv/libuv/issues
       > ======================================================
       > make[1]: *** [Makefile:5613: check-TESTS] Error 1
       > make[1]: Leaving directory '/build/source'
       > make: *** [Makefile:5878: check-am] Error 2
       For full logs, run 'nix log /nix/store/gcasfqr3rkfhwv3f9yf8iyjf8lpki5mx-libuv-static-x86_64-unknown-linux-musl-1.44.2.drv'.
```
</details>


### Additional context

https://github.com/libuv/libuv/issues/3706

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

I was instructed to add a patch in the related issue and run the test, which I did like so

1. Find the source for the derivation that was failing and copy it somewhere writeable

```
$ nix repl "https://github.com/NixOS/nixpkgs/archive/23028550cf3064f519bb302361c9500ebc837f92.tar.gz"
warning: future versions of Nix will require using `--file` to load a file
Welcome to Nix 2.10.3. Type :? for help.

Loading installable ''...
Added 17089 variables.
nix-repl> pkgsStatic.libuv.src.outPath
"/nix/store/61k994863013giis21bfzwwsf6nr9l87-source"
nix-repl> :q

$ cp -R /nix/store/61k994863013giis21bfzwwsf6nr9l87-source /tmp/libuv
$ nix-shell "https://github.com/NixOS/nixpkgs/archive/23028550cf3064f519bb302361c9500ebc837f92.tar.gz" -A pkgsStatic.libuv
$ cd /tmp/
$ chmod -R u+rw libuv
$ sh autogen.sh
$ ./configure
$ make -j
```

But I got a different error:

```

/nix/store/c3qpvp6s9bbskqsay1q3snabqls544w7-x86_64-unknown-linux-musl-binutils-2.38/bin/x86_64-unknown-linux-musl-ld: /nix/store/ym1418bazp23rqg9l43kqzhvv2cdibby-x86_64-unknown-linux-musl-stage-final-gcc-11.3.0/lib/gcc/x86_64-unknown-linux-musl/11.3.0/crtbeginT.o: relocation R_X86_64_32 against hidden symbol `__TMC_END__' can not be used when making a shared object
/nix/store/c3qpvp6s9bbskqsay1q3snabqls544w7-x86_64-unknown-linux-musl-binutils-2.38/bin/x86_64-unknown-linux-musl-ld: failed to set dynamic section sizes: bad value
collect2: error: ld returned 1 exit status
make: *** [Makefile:1784: libuv.la] Error 1
```

I guess this means my nix-shell environment is different from the one with nix-build somehow.

I'm not sure how to further debug this.

--- a/pkgs/development/libraries/libuv/default.nix
+++ b/pkgs/development/libraries/libuv/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, lib, fetchFromGitHub, autoconf, automake, libtool, pkg-config, ApplicationServices, CoreServices }:
+{ stdenv, lib, fetchFromGitHub, fetchpatch, autoconf, automake, libtool, pkg-config, ApplicationServices, CoreServices, pkgsStatic }:
 
 stdenv.mkDerivation rec {
   version = "1.44.2";
@@ -10,6 +10,15 @@ stdenv.mkDerivation rec {
     rev = "v${version}";
     sha256 = "sha256-K6v+00basjI32ON27ZjC5spQi/zWCcslDwQwyosq2iY=";
   };
+
+  patches = [
+    # Fix tests for statically linked variant upstream PR is
+    # https://github.com/libuv/libuv/pull/3735
+    (fetchpatch {
+      url = "https://github.com/libuv/libuv/commit/9d898acc564351dde74e9ed9865144e5c41f5beb.patch";
+      sha256 = "sha256-6XsjrseD8a+ny887EKOX0NmHocLMXGf2YL13vkNHUZ0=";
+    })
+  ];
 
   postPatch = let
     toDisable = [
@@ -69,6 +78,8 @@ stdenv.mkDerivation rec {
 
   # Some of the tests use localhost networking.
   __darwinAllowLocalNetworking = true;
+
+  passthru.tests.static = pkgsStatic.libuv;
 
   meta = with lib; {
     description = "A multi-platform support library with a focus on asynchronous I/O";

